test/test_mps.py

```diff
@@ -1223,6 +1221,67 @@ def test_slice(self):
         mps_slice4 = mps_x[1, :].to('cpu')
         self.assertEqual(cpu_slice4, mps_slice4)
 
+    def test_slice_contiguous_view(self):
+        # https://github.com/pytorch/pytorch/issues/77750
+
+        def helper(operator):
+            t_mps = torch.tensor([1, 2, 3, 4], device="mps")
+            t_cpu = torch.tensor([1, 2, 3, 4], device="cpu")
+
+            # contiguous view
+            x_mps = t_mps[2:]  # 3, 4
+            y_mps = t_mps[:2]  # 1, 2
+
+            x_cpu = t_cpu[2:]
+            y_cpu = t_cpu[:2]
+
+            res_mps = res_cpu = None
+            if operator == "<=":
```

Comment:
It would be cleaner to use the magic methods directly to avoid all these if statements:
```python
# To do "<="
op = "__lt__"
res_mps = getattr(x_mps, op)(y_mps)
```
You can see the python doc for the exact name for all these methods: https://docs.python.org/3/reference/datamodel.html#object.__lt__